File: src/main/java/dsl/BranchMapping.java
package dsl;

import java.util.ArrayList;
import java.util.List;

public class BranchMapping implements MappingRule {
    private final List<ConditionalBranch> branches;

    BranchMapping(List<ConditionalBranch> branches) {
        this.branches = new ArrayList<>(branches);
    }

    @Override
    public void apply(MappingContext context, Object target) {
        for (ConditionalBranch branch : branches) {
            if (branch.getCondition().test(context.getSourceData())) {
                for (MappingRule rule : branch.getActions()) {
                    rule.apply(context, target);
                }
                break; // Stop after first matching branch
            }
        }
    }
}

File: src/main/kotlin/xyz/mahmoudahmed/dsl/conditional/BranchMapping.kt
package xyz.mahmoudahmed.dsl.conditional

import xyz.mahmoudahmed.dsl.core.MappingContext
import xyz.mahmoudahmed.dsl.core.MappingRule
import java.util.ArrayList

/**
 * Implements conditional branching in the mapping DSL.
 * Allows for different mapping rules to be applied based on conditions.
 */
class BranchMapping(
    branches: List<ConditionalBranch>
) : MappingRule {
    private val branches = ArrayList(branches)

    override fun apply(context: MappingContext, target: Any) {
        for (branch in branches) {
            if (branch.condition(context.sourceData)) {
                for (rule in branch.actions) {
                    rule.apply(context, target)
                }
                break
            }
        }
    }
}
